File: 백준/Silver/10845. 큐/큐.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayDeque;
import java.util.Deque;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        Deque<Integer> queue = new ArrayDeque<>();
        StringBuilder sb = new StringBuilder();
        for(int i=0; i<n; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            switch (st.nextToken()) {
                case "push":
                    int num = Integer.parseInt(st.nextToken());
                    queue.offer(num);
                    break;
                case "pop":
                    if(queue.isEmpty()) sb.append(-1 + "\n");
                    else sb.append(queue.pop() + "\n");
                    break;
                case "size":
                    sb.append(queue.size() + "\n");
                    break;
                case "empty":
                    if(queue.isEmpty()) sb.append(1 + "\n");
                    else sb.append(0 + "\n");
                    break;
                case "front":
                    if(queue.isEmpty()) sb.append(-1 + "\n");
                    else sb.append(queue.peekFirst() + "\n");
                    break;
                case "back":
                    if(queue.isEmpty()) sb.append(-1 + "\n");
                    else sb.append(queue.peekLast() + "\n");
                    break;
                default:
                    break;
            }
        }
        System.out.println(sb);
    }
}

File: 백준/Silver/10845. 큐/큐.kt
import java.io.*

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val sb = StringBuilder()
    val n = br.readLine().toInt()
    val queue = ArrayDeque<Int>()

    repeat(n) {
        val input = br.readLine().split(" ")
        when (input[0]) {
            "push" -> queue.addFirst(input[1].toInt())
            "pop" -> sb.appendLine(if (queue.isEmpty()) -1 else queue.removeLast())
            "size" -> sb.appendLine(queue.size)
            "empty" -> sb.appendLine(if (queue.isEmpty()) 1 else 0)
            "front" -> sb.appendLine(if (queue.isEmpty()) -1 else queue.last())
            "back" -> sb.appendLine(if (queue.isEmpty()) -1 else queue.first())
        }
    }

    print(sb)
}